Shm management

ktf opened the pull request.

ktf committed:
DPL: do own management of shared memory

We do not spawn a shm-monitor because that will create zombies when launched
inside a docker container. Instead by default we cleanup the shared memory
session on exit. Default behavior can still be obtained by passing
--shm-monitor true.

sawenzel committed:
Digitization: Optional MCTruth propagation TPC/ITS/MFT

Allow to disable MC labels in TPC/ITS/MFT digitization.
Add mctruth flag to DigiParams.

This commit allows to reach a timeframe with >= 500PbPb.

In a future commit, the actual creation of labels within
the digit process function can be disabled in addition.
For the moment, merely the messaging and IO part has been disabled.

shahor02 committed:
Set DataHeader.firstTForbit in RawFileReaderWorkflow (#3573)

Co-authored-by: shahoian <ruben.shahoyan@cern.ch>

ktf committed:
DPL: adapt to work with FairLogger 1.6 (#3574)

ktf commented:
@matthiasrichter you will need also this to avoid killing the test machines with zombies.

matthiasrichter commented:
Which one is the relevant commit? I've tested with the merged branch but get error
```
[ERROR] error while setting up workflow: Error while parsing serialised workflow
```
so something got lost. I have fixed that a while ago.

For the moment I've just removed the commit enabling shm

ktf committed:
DPL: make sure the RegionInfoCallback is actually invoked (#3579)

Apparently our service registry gets confused if it gets a non-const
object.

matthiasrichter committed:
DPL: Base validity check for InputRecord route on header message only (#3578)

FairMQ shared memory transport gives a nullptr for a zero-length message,
while zmq transport returns a valid pointer. Validity check did require
both valid header and payload message pointers. That makes zero length
shared memory messages to be masked out.

In principle it's enough to check for the header message pointer because
the DataRelayer logic makes sure that the message pair has arrived. To
check for the payload message pointer we would need to check the payload
size and her it would mean to get DataHeader and check. Some overhead
we probably do not want.

sawenzel committed:
Optional MCTruh handling for FDD digitization

hahassan7 committed:
[EMCAL-630] DPL component for Raw to Cell conversion (#3079)

-DPL component for Raw to Cell conversion

mpuccio committed:
Fix copy/paste typo in starting point definition (#3583)
